libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -273,14 +270,15 @@ static bool _topicFilterMatch( const char * pTopicName,
         filterIndex++;
     }
 
-    /* If the end of both strings has been reached, they match. */
-    if( ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength ) )
+    if( status == false )
```

Comment:
Is this condition necessary? Is there any harm in setting status to `true` in the event when status is already `true`?

Reply:
There is no harm, but if status is true, then we don't want to do unnecessary  comparison.